client/components/external-link/with-tracking.jsx

```diff
@@ -0,0 +1,57 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import ExternalLink from './index';
+import { recordTracksEvent } from 'state/analytics/actions';
+
+class ExternalLinkWithTracking extends Component {
+	handleClickEvent() {
+		const { onClick, tracksEventName, tracksEventProps } = this.props;
+
+		this.props.recordTracksEvent( tracksEventName, tracksEventProps );
+
+		if ( onClick ) {
+			onClick();
+		}
+	}
+
+	render() {
+		const {
+			onClick,
+			recordTracksEvent: recordEvent,
+			tracksEventName,
+			tracksEventProps,
+			...props
+		} = this.props;
+
+		return <ExternalLink onClick={ () => this.handleClickEvent() } { ...props } />;
```

Comment:
Let's define this arrow function as part of `handleClickEvent` because now it gets defined on every re-render. I'd very much prefer it's defined on every call instead. So `handleClickEvent` would become:

```
handleClickEvent() {
	return () => {
		// current contents of `handleClickEvent`
	}
}
```